web/src/compose_actions.js

```diff
@@ -270,6 +270,12 @@ export function start(msg_type, opts) {
 
     compose_state.set_message_type(msg_type);
```

Comment:
I think it might be fruitful to move this call above the `opts.content` check, and have your check for this feature sit as an else clause to `opts.content !== undefined`, which is the code path used by "quote and reply" and similar features, which I expect should always .

Hmm, `compose_reply.quote_and_reply` doesn't use that feature.

Reply:
@timabbott FYI waiting on your response to this to move forward

Reply:
This feels very messy. I think it might be better to just have an option passed in the small number of code paths where the intent is to start with a fresh messy, rather than adding an option to all of the places we open compose?

Reply:
> where the intent is to start with a fresh messy

Is this supposed to say fresh draft? As in, only pass the option when we _don't_ want to load a previous draft?

Yeah I agree it's pretty messy, which I don't like. The reason I ended up doing it this way is that it's not always obvious that drafts are being overwritten, so future situations where we add code to open the compose box and then overwrite it, it'll be really important to make sure we're not loading a previous draft. If loading a previous draft happens by default, then it just makes it easier for that to happen.

But I'm happy to switch it to only pass the option sometimes, if that seems like the better tradeoff. I just want to be clear about the tradeoff.

Reply:
Yeah, "fresh draft". Correct, we'd pass the option when we don't want to load a draft. I think there's be at most a handful of places that needs to be passed.

I'm not sure I understand the point about drafts being overwritten -- if you see a draft in the compose box and replace it with other text, I think you'd expect that the old text is wrong. E.g. gmail drafts work the same way.

Reply:
The overwriting doesn't happen from user choice, but from code.

If we forgot to pass this option for quote-replying, and there was a draft we could auto-load, then that draft would automatically be overwritten when we replaced the compose content with the quote reply, and the draft would be lost. So it's very important to remember to pass the option for starting a fresh draft when it's relevant, and if we forget then drafts can get overwritten (essentially deleted).